src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -9667,201 +9671,206 @@ void WaterThermalTankData::CalcHeatPumpWaterHeater(EnergyPlusData &state, bool c
     // set water-side mass flow rate for final calculation
     state.dataLoopNodes->Node(HPWaterInletNode).MassFlowRate = MdotWater * state.dataWaterThermalTanks->hpPartLoadRatio;
 
-    if (MaxSpeedNum > 0) {
+    if (HeatPump.dxCoilAvailSched->getCurrentVal() > 0) {
```

Comment:
If the HPWH DX Coil is scheduled OFF skip the DX coil calculations and set the compressor PLR to `0.0`.